Prep grpc client generation

wba2hi opened the pull request:
Preparation for the gRPC Client Generation.

This PR tries to align operations which are necessary to support a flawless gRPC Client Code generation as well as the integration required into a custom app. 

- Adding a LogcatLoggingStrategy so that Android Apps can log messages
- Add a VehicleApplication which provides access to the middleware
- Add a VehicleApp (which implements) VehicleApplication, which is used inside the VehicleAppService to access VehicleApplication functionality
- Update kuksa.val.v2 protocol to the latest version
- Add a Main Pipeline
- Align Proto Files used

wba2hi committed:
Add Tag and Verbose to Logger

(cherry picked from commit c8999eb9edbffa6a41244c339ba49f9a3236fb19)

# Conflicts:
#	app/src/main/java/com/example/service/GrpcCarService.kt

wba2hi committed:
Update kuksa.val.v2 proto

wba2hi committed:
Add LogcatLoggingStrategy

(cherry picked from commit 11a7e8c2e8e63773e30dc736698f6deea76b0785)

wba2hi committed:
Add Custom VehicleApp Class

(cherry picked from commit c16219626daee0a31493874d83e03744b1f2e738)

wba2hi committed:
Add PUML Documentation

wba2hi committed:
Rename App to SampleApp

wba2hi committed:
Add Main Pipeline

wba2hi committed:
Adapt Proto Files

Chrylo commented:
LGTM  for an initial setup. 

Only thing I found:
- Java compatibility with the newest Android Studio Ladybug creates issues because Ladybug uses Java 21 now instead of 17 bundled within.

wba2hi committed:
Align Java and Kotlin JavaVersion

Increased Sample App JavaVersion from Java 8 to
Java 11 because Java 11 is slowly fading out.

wba2hi committed:
Align some IntelliJ Files

wba2hi committed:
Adapt Build Pipeline to build with Java 21

wba2hi committed:
Add missing Import

doosuu reviewed (approved):
lgtm
